subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CopyPermissionsIntegrationTest.groovy

```diff
@@ -193,5 +194,32 @@ class CopyPermissionsIntegrationTest extends AbstractIntegrationSpec {
         testTargetFile.canWrite()
     }
 
+    @Requires(TestPrecondition.FILE_PERMISSIONS)
```

Comment:
We should add some test coverage for what happens when a file or directory cannot be read and is _not_ excluded, making sure that the user receives a reasonable error message explaining the problem.

Reply:
Thanks for pointing out that, and this was also mentioned by @jjohannes . I had a simple unit test for this scenario: [https://github.com/blindpirate/gradle/blob/f540e6a84ec4bbb0910853b2ac2cea67686d3637/subprojects/core/src/test/groovy/org/gradle/api/internal/file/collections/jdk7/Jdk7UnauthorizedDirectoryWalkerTest.groovy#L67](https://github.com/blindpirate/gradle/blob/f540e6a84ec4bbb0910853b2ac2cea67686d3637/subprojects/core/src/test/groovy/org/gradle/api/internal/file/collections/jdk7/Jdk7UnauthorizedDirectoryWalkerTest.groovy#L67), and look like we need a more reasonable error message instead the default `AccessDeniedException`.